src/tests/test_common/homestore_test_common.hpp

```diff
@@ -190,14 +192,35 @@ class HSTestHelper {
         }
 
         std::vector< homestore::dev_info > device_info;
-        if (SISL_OPTIONS.count("device_list")) {
-            s_dev_names = SISL_OPTIONS["device_list"].as< std::vector< std::string > >();
+        if (!cust_dev_names.empty() || SISL_OPTIONS.count("device_list")) {
+            cust_dev_names.empty() ? s_dev_names = SISL_OPTIONS["device_list"].as< std::vector< std::string > >()
+                                   : s_dev_names = std::move(cust_dev_names);
             LOGINFO("Taking input dev_list: {}",
                     std::accumulate(
                         s_dev_names.begin(), s_dev_names.end(), std::string(""),
                         [](const std::string& ss, const std::string& s) { return ss.empty() ? s : ss + "," + s; }));
 
+            // zero the homestore pdev's first block for each device;
+            auto const zero_size = hs_super_blk::first_block_size();
+            std::vector< int > zeros(zero_size, 0);
+            for (auto const& d : cust_dev_names) {
+                if (!std::filesystem::exists(d)) {
+                    LOGINFO("Device {} does not exist", d);
+                    HS_REL_ASSERT(false, "Device does not exist");
+                }
+
+                auto fd = ::open(d.c_str(), O_RDWR | O_CREAT, 0640);
```

Comment:
is the O_CREATE flag causing a regular file being created that covers the real device?

Reply:
good point, however this code is copied from iomgr during homestore init. 
e.g. open flag is set by homestore here, where O_CREATE is set for all cases (except for readonly): https://github.com/eBay/HomeStore/blob/master/src/lib/device/device_manager.cpp#L38

Reply:
verified and seems O_CREATE here doesn't make any difference, and also looked at 1.3 code it also passes O_CREATE flag.